### PR TITLE
Fix readability-const-return-type, readability-delete-null-pointer, readability-simplify-boolean-expr

### PR DESCRIPTION
**Pull request by veblush**
Fix clang-tidy:
- readability-const-return-type
- readability-delete-null-pointer
- readability-simplify-boolean-expr

### CODE REVIEW

File: src/core/lib/gpr/sync_abseil.cc [hunk 1]
```diff
@@ -62,7 +62,7 @@ void gpr_mu_unlock(gpr_mu* mu) ABSL_NO_THREAD_SAFETY_ANALYSIS {
 
 int gpr_mu_trylock(gpr_mu* mu) {
   GPR_TIMER_SCOPE("gpr_mu_trylock", 0);
-  int ret = reinterpret_cast<absl::Mutex*>(mu)->TryLock() == true;
+  int ret = reinterpret_cast<absl::Mutex*>(mu)->TryLock();
   return ret;
```
Comment (drfloob) [inline src/core/lib/gpr/sync_abseil.cc:66]: I think this could be simplified to a one-liner.

File: src/core/lib/gpr/sync_abseil.cc [hunk 2]
```diff
@@ -90,8 +90,7 @@ int gpr_cv_wait(gpr_cv* cv, gpr_mu* mu, gpr_timespec abs_deadline) {
   timespec ts = {static_cast<decltype(ts.tv_sec)>(abs_deadline.tv_sec),
                  static_cast<decltype(ts.tv_nsec)>(abs_deadline.tv_nsec)};
   int ret = reinterpret_cast<absl::CondVar*>(cv)->WaitWithDeadline(
-                reinterpret_cast<absl::Mutex*>(mu),
-                absl::TimeFromTimespec(ts)) == true;
+      reinterpret_cast<absl::Mutex*>(mu), absl::TimeFromTimespec(ts));
   return ret;
```
Comment (drfloob) [inline src/core/lib/gpr/sync_abseil.cc:94]: same here

File: src/cpp/common/auth_property_iterator.cc [hunk 3]
```diff
@@ -61,7 +61,7 @@ bool AuthPropertyIterator::operator!=(const AuthPropertyIterator& rhs) const {
   return !operator==(rhs);
 }
 
-const AuthProperty AuthPropertyIterator::operator*() {
+AuthProperty AuthPropertyIterator::operator*() {
   return std::pair<grpc::string_ref, grpc::string_ref>(
```
Comment (drfloob) [inline src/cpp/common/auth_property_iterator.cc:65]: Interesting design! Ignore my earlier comment, return by value was intended here, so the const is superfluous. string_refs appear to be immutable by design as well.

**Review by drfloob — APPROVED** [review]
LGTM. Auto-merge disabled to accommodate the two fixit comments.

**Review by markdroth — APPROVED** [review]

File: src/core/lib/gpr/sync_abseil.cc (hunk 1 above)
Comment (veblush) [reply src/core/lib/gpr/sync_abseil.cc]: done.

File: src/cpp/common/auth_property_iterator.cc (hunk 3 above)
Comment (veblush) [reply src/cpp/common/auth_property_iterator.cc:65]: Yeah, you're right.

**Pull request merged**